core/core.go

```diff
@@ -152,7 +153,7 @@ type Mounts struct {
 	Ipns mount.Mount
 }
 
-func (n *IpfsNode) startOnlineServices(ctx context.Context, routingOption RoutingOption, hostOption HostOption, do DiscoveryOption, pubsub, mplex bool) error {
+func (n *IpfsNode) startOnlineServices(ctx context.Context, routingOption RoutingOption, hostOption HostOption, do DiscoveryOption, bo DiscoveryOption, pubsub, mplex bool) error {
```

Comment:
Any reason to not just pass in an array of these? That would allow us to avoid changing this code again.